File: leetcode/100-999/169. Majority Element/solution_3.cs
public class Solution 
{
    public int MajorityElement(int[] nums) 
    {
        int majority = -111;
        int count = 0;

        foreach (int num in nums)
        {
            if (count == 0)
                majority = num;

            count += majority == num ? 1: -1;
        }

        return majority;
    }
}

File: leetcode/100-999/169. Majority Element/solution_3.js
/**
 * @param {number[]} nums
 * @return {number}
 */

var majorityElement = function(nums) 
{
    let majority = -111
    let count = 0    

    for (const num of nums)
    {
        if (count == 0)
            majority = num

        majority == num 
            ? ++count
            : --count
    }

    return majority
};